dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+# standard imports
+import logging
+import uuid
+import time
+import os
+import json
+from datetime import datetime
+
+# internal imports
+from dripline import core
+from dragonfly.implementations import EthernetProvider
+
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+__all__.append('PsyllidProvider')
+class PsyllidProvider(core.Provider, core.Spime):
```

Comment:
Also please convert hard tabs to soft tabs. If you use vi(m) then `:%s/\t/    /g` will do it (but you'll need to make sure you don't break any indentation blocks).